test/unit/spec/userMenuSpec.js

```diff
@@ -65,29 +67,86 @@ define(['jquery', 'userMenu', 'narrativeConfig', 'narrativeMocks'], (
             });
         });
 
-        it('Should trigger a logout popup when clicking the logout button', () => {
-            const $elem = $('<div>');
-            const userMenu = makeTestUserMenu($elem);
-            return userMenu.start().then(() => {
-                expect(
-                    document.querySelector('.modal [data-element="signout-warning-body"]')
-                ).toBeNull();
-                jasmine.clock().install();
-                // click the signout button
-                $elem.find('#signout-button').click();
-                jasmine.clock().tick(2000);
-                // expect to see the modal appear now
-                expect(
-                    document.querySelector('.modal [data-element="signout-warning-body"]')
-                ).not.toBeNull();
-                // click the cancel button (it has the default button style)
-                document.querySelector('a[data-element="cancel-signout"]').click();
-                jasmine.clock().tick(2000);
-                // expect the modal to go away
-                expect(
-                    document.querySelector('.modal [data-element="signout-warning-body"]')
-                ).toBeNull();
-                jasmine.clock().uninstall();
+        // logout popups
+        const logoutTests = [
+            {
+                desc: 'simulate cancelling signout',
+                button: 'a[data-element="cancel-signout"]',
+                // there should be no 'logout.kbase' event
+                finalTest: (callArgs) => {
+                    return callArgs.every((args) => {
+                        return typeof args[0] !== 'string' || args[0] !== 'logout.kbase';
+                    });
+                },
+            },
+            {
+                desc: 'simulate signout',
+                button: 'a[data-element="signout"]',
+                // the 'logout.kbase' event should have been triggered
+                finalTest: (callArgs) => {
+                    return callArgs.some((args) => {
+                        return (
+                            typeof args[0] === 'string' &&
+                            args[0] === 'logout.kbase' &&
+                            args[1] === true
+                        );
+                    });
+                },
+            },
+        ];
+
+        logoutTests.forEach((test) => {
```

Comment:
Fixed the modal issue and also added in an extra test -- this test now simulates clicking the logout button, and then choosing either 'cancel logout' or 'logout'. Events are captured and checked to ensure that the correct action took place.